src/overlays/effects/ovl_Effect_Ss_En_Fire/z_eff_ss_en_fire.c

```diff
@@ -5,23 +5,144 @@
+    camYaw = (Camera_GetCamDirYaw(GET_ACTIVE_CAM(play)) + 0x8000);
+    Matrix_RotateYS(camYaw, MTXMODE_APPLY);
+
+    scale = Math_SinS(this->life * 0x333) * (this->rScale * 0.00005f);
+    Matrix_Scale(scale, scale, scale, MTXMODE_APPLY);
+    gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+
+    redGreen = this->life - 5;
+
+    if (redGreen < 0) {
+        redGreen = 0;
+    }
+
+    func_8012C2DC(play->state.gfxCtx);
+    gDPSetEnvColor(POLY_XLU_DISP++, redGreen * 12.7f, 0, 0, 0);
+    gDPSetPrimColor(POLY_XLU_DISP++, 0x0, 0x80, redGreen * 12.7f, redGreen * 12.7f, 0, 255);
+    gSPSegment(
+        POLY_XLU_DISP++, 0x08,
+        Gfx_TwoTexScroll(play->state.gfxCtx, 0, 0, 0, 0x20, 0x40, 1, 0, (this->rScroll * -20) & 0x1FF, 0x20, 0x80));
+
+    if ((this->rFlags & 0x7FFF) || (this->life < 18)) {
+        gSPDisplayList(POLY_XLU_DISP++, gEffFire2DL);
+    } else {
+        gSPDisplayList(POLY_XLU_DISP++, gEffFire1DL);
+    }
+
+    CLOSE_DISPS(gfxCtx);
+}
+
+typedef struct {
+    /* 0x000 */ Actor actor;
+    /* 0x14C */ Vec3f firePos[10];
+} FireActorF; // size = 0x1BC
+
+typedef struct {
+    /* 0x000 */ Actor actor;
+    /* 0x14C */ Vec3s firePos[10];
+} FireActorS; // size = 0x180
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Effect_Ss_En_Fire/EffectSsEnFire_Init.s")
+void EffectSsEnFire_Update(PlayState* play, u32 index, EffectSs* this) {
+    this->rScroll++;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Effect_Ss_En_Fire/EffectSsEnFire_Draw.s")
+    if (this->actor != NULL) {
+        if (this->actor->colorFilterTimer >= 22) {
+            this->life++;
+        }
+        if (this->actor->update != NULL) {
+            Math_SmoothStepToS(&this->rScale, this->rScaleMax, 1, this->rScaleMax >> 3, 0);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Effect_Ss_En_Fire/EffectSsEnFire_Update.s")
+            if (this->rBodyPart < 0) {
+                Matrix_Translate(this->actor->world.pos.x, this->actor->world.pos.y, this->actor->world.pos.z,
+                                 MTXMODE_NEW);
+                Matrix_RotateYS(this->rYaw + this->actor->shape.rot.y, MTXMODE_APPLY);
+                Matrix_RotateXS(this->rPitch + this->actor->shape.rot.x, MTXMODE_APPLY);
+                Matrix_MultVec3f(&this->vec, &this->pos);
+            } else if (this->rFlags & ENFIRE_FLAGS_BODYPART_POS_VEC3S) {
+                this->pos.x = ((FireActorS*)this->actor)->firePos[this->rBodyPart].x;
+                this->pos.y = ((FireActorS*)this->actor)->firePos[this->rBodyPart].y;
+                this->pos.z = ((FireActorS*)this->actor)->firePos[this->rBodyPart].z;
+            } else {
+                this->pos.x = ((FireActorF*)this->actor)->firePos[this->rBodyPart].x;
+                this->pos.y = ((FireActorF*)this->actor)->firePos[this->rBodyPart].y;
+                this->pos.z = ((FireActorF*)this->actor)->firePos[this->rBodyPart].z;
```

Comment:
Can't be temps, I suppose?

Reply:
At least not in the obvious way. If you have any ideas here is a scratch https://decomp.me/scratch/zzu0H

Note1: this functionality is just leftover from OoT, I remember seeing where this effect gets spawned, and no actors actually specify a bodypart.

Note2: this is one of the main functions where it makes sense `this->actor` is actually an `Actor*` and not some generic `void*` since an `Actor*` temp here also does not seem to work.